src/functionalTest/java/uk/gov/hmcts/reform/pbis/SubscriptionProcessingTest.java

```diff
@@ -0,0 +1,137 @@
+package uk.gov.hmcts.reform.pbis;
+
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+import static org.assertj.core.api.Java6Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+
+import com.microsoft.azure.servicebus.primitives.ServiceBusException;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.awaitility.Duration;
+import org.junit.Before;
+import org.junit.Test;
+import uk.gov.hmcts.reform.pbis.model.PrivateBetaRegistration;
+import uk.gov.hmcts.reform.pbis.utils.ServiceBusFeeder;
+import uk.gov.service.notify.Notification;
+import uk.gov.service.notify.NotificationClient;
+import uk.gov.service.notify.NotificationClientException;
+
+
+public class SubscriptionProcessingTest {
+
+    private static final Configuration config = new Configuration();
+
+    private static final Duration MAX_WAIT_TIME_MS = new Duration(
+        config.getServiceBusPollingDelayInMs() + 2000,
+        MILLISECONDS
+    );
+
+    private ServiceBusFeeder serviceBusFeeder;
+    private NotificationClient notificationClient;
+
+    @Before
+    public void setUp() throws ServiceBusException, InterruptedException {
+        serviceBusFeeder = new ServiceBusFeeder(config.getServiceBusConnectionString());
+        notificationClient = new NotificationClient(config.getNotifyApiKey());
+    }
+
+    @Test
+    public void should_send_welcome_email_when_valid_message_present() throws Exception {
+
+        PrivateBetaRegistration registration = getSampleRegistration(config.getServiceName());
+
+        serviceBusFeeder.sendMessage(registration);
+
+        waitUntilRegistrationIsProcessed(registration.referenceId);
+
+        List<Notification> emails = getSentEmails(registration.referenceId);
+
+        assertThat(emails).hasSize(1);
+        Notification email = emails.get(0);
+
+        assertThat(email).isNotNull();
+        assertThat(email.getEmailAddress()).isEqualTo(Optional.of(registration.emailAddress));
+
+        assertThat(email.getNotificationType()).isEqualTo("email");
+        assertThat(email.getTemplateId().toString()).isEqualTo(config.getTemplateId());
+        assertThat(email.getBody()).contains("First name: " + registration.firstName);
+        assertThat(email.getBody()).contains("Last name: " + registration.lastName);
+        assertThat(email.getBody()).contains("Welcome link: " + config.getWelcomeLink());
```

Comment:
it may be worth using soft assertions here 

http://joel-costigliola.github.io/assertj/assertj-core-features-highlight.html#soft-assertions